source/shared/cpp/ObjectModel/TableColumnDefinition.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "TableColumnDefinition.h"
+#include "Util.h"
+
+namespace AdaptiveSharedNamespace
+{
+    TableColumnDefinition::TableColumnDefinition() :
+        m_horizontalCellContentAlignment(HorizontalAlignment::Left),
+        m_verticalCellContentAlignment(VerticalAlignment::Top), m_pixelWidth{}, m_width{}
+    {
+    }
+
+    std::string TableColumnDefinition::Serialize() { return ParseUtil::JsonToString(SerializeToJsonValue()); }
+
+    Json::Value TableColumnDefinition::SerializeToJsonValue()
+    {
+        Json::Value root;
+
+        if (m_horizontalCellContentAlignment != HorizontalAlignment::Left)
+        {
+            root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::HorizontalCellContentAlignment)] =
+                HorizontalAlignmentToString(m_horizontalCellContentAlignment);
+        }
+
+        if (m_verticalCellContentAlignment != VerticalAlignment::Top)
+        {
+            root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::VerticalCellContentAlignment)] =
+                VerticalAlignmentToString(m_verticalCellContentAlignment);
+        }
+
+        if (m_width.has_value())
+        {
+            root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Width)] = m_width.value();
+        }
+
+        if (m_pixelWidth.has_value())
+        {
+            std::ostringstream pixelStr;
+            pixelStr << m_pixelWidth.value() << "px";
```

Comment:
value_or()

Reply:
going to try using `operator*` instead of `value()` in case that works on iOS -- see 24dbfdd